src/main/webapp/index-script.js

```diff
@@ -1,16 +1,74 @@
+var map;
+/* Loades page and main buttons. */
 function loadPage() {
     createMap();
     loadMainButtons();
 }
 
-// Creates a map centered at the Googleplex!
+/* Creates a map centered at the Googleplex! .*/
 function createMap() {
-  const map = new google.maps.Map(
-      document.getElementById('map'),
-      {center: {lat: 37.422, lng: -122.0841}, zoom: 13, mapTypeId: 'satellite'}); 
+  map = new google.maps.Map(
+    document.getElementById('map'),
+    {center: {lat: 37.422, lng: -122.0841}, zoom: 13,
+    mapTypeControlOptions: {mapTypeIds: ['roadmap']}});
+
+  // Create the initial InfoWindow.
+  var infoWindow = new google.maps.InfoWindow(
+      {content: 'Open javascript console (ctrl + shift + j) then click the map to see the placeIDs of nearby locations (within 50m)',
+       position: {lat: 37.422, lng: -122.0841}});
+  infoWindow.open(map);
+
+  // Configure the click listener.
+  map.addListener('click', function(mapsMouseEvent) {
+    // Close the current InfoWindow.
+    infoWindow.close();
+
+    searchByCoordinates(mapsMouseEvent.latLng);
+  });
+}
+
+/* Takes coordinates from mouseclick and searches for locations within a given radius of those coordinates. */
+function searchByCoordinates(coordinate) {
+  var request = {
+    location: coordinate,
+    radius: '50',
+    rankBy: google.maps.places.RankBy.PROMINENCE
+  };
+
+  var service = new google.maps.places.PlacesService(map);
+  // Print resulting IDs to log.
+  service.nearbySearch(request, (results, status) => {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+      results.forEach((result)=> {
+        var request = {
```

Comment:
You have two variables named request; perhaps change to be more meaningful?

Reply:
Agreed, should be a non-issue when handling the results in the next branch, but i'll be more careful.